state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -309,6 +315,15 @@ pub mod pallet {
 	#[pallet::getter(fn aborted_broadcasts)]
 	pub type AbortedBroadcasts<T, I = ()> = StorageValue<_, BTreeSet<BroadcastId>, ValueQuery>;
 
+	#[pallet::storage]
+	#[pallet::getter(fn incoming_key_broadcast_id)]
+	pub type IncomingKeyAndBroadcastId<T, I = ()> =
+		StorageValue<_, (AggKey<T, I>, BroadcastId), OptionQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn current_on_chain_key)]
+	pub type CurrentOnChainKey<T, I = ()> = StorageValue<_, AggKey<T, I>, OptionQuery>;
```

Comment:
I would leave a comment here explaining why this is necessary, in particular considering we have the key in the threshold signature pallet (which is not necessarily on the external chain since we optimistically rotate, but worth making that clear here).